### PR TITLE
Add HTTP Method GET Support for RETS Transactions.

### PR DESCRIPTION
**Pull request by erinus**
Add HTTP Method GET Support for RETS Transactions.

Tested POST support successfully with CRMLS.
Tested GET support successfully with SFARMLS.

SFARMLS use only GET for Login Transaction, but GET and POST are both supported for GetMetadata / GetObject.

### CODE REVIEW

File: rets/http/client.py [hunk 1]
```diff
@@ -286,6 +307,32 @@ def _url_for(self, transaction: str) -> str:
...
+        })
+
+        if payload is not None:
+            url = '%s?%s' % (url, urlencode(payload))
+
+        response = self._session.get(url, headers=headers, auth=self._http_auth)
+
+        response.raise_for_status()
+
+        if not self._rets_session_id:
```
[the remark this answers is not in the document]
Comment (erinus) [reply rets/http/client.py]: When GET request happens at first time for Login Transaction, digest in RETS-UA-Authorization will calculate with empty rets_session_id (and it initialized at line 65). After login succesfully, rets_session_id is not empty, so digest in RETS-UA-Authorization must be calculated again.

I will add comments with my modification. Thank you for your advice.

File: rets/http/client.py [hunk 2]
```diff
@@ -27,21 +27,24 @@ def __init__(self,
                  user_agent: str = 'rets-python/0.3',
                  user_agent_password: str = None,
                  rets_version: str = '1.7.2',
+                 use_post_method: bool = True,
```
[the remark this answers is not in the document]
Comment (erinus) [reply rets/http/client.py]: It's okey. Your project is excellent. I just wanna help it to fit more RETS source. You can do any modification to my commit. Thanks.

**Comment by erinus** [comment]
Sould I do some modification for merge approving? Could you give me guidance to pass it? Thanks.

**Comment by blakeembrey** [comment]
@erinus Let me review and land this tomorrow for you 😄

**Comment by blakeembrey** [comment]
@erinus Sorry for the delay, I've picked your branch and make additional changes in https://github.com/opendoor-labs/rets/pull/44. Please let me know if you'd prefer I didn't land the changes since you closed this PR.

**Pull request closed without merging**